Number_types/include/CGAL/Number_types/internal/Exact_type_selector.h

```diff
@@ -54,19 +54,24 @@ namespace CGAL { namespace internal {
 // It should support the built-in types.
 template < typename >
 struct Exact_field_selector
+
 #ifdef CGAL_USE_GMPXX
 { typedef mpq_class Type; };
 #elif defined(CGAL_USE_GMP)
-# if defined(CGAL_USE_BOOST_MP)
-{ typedef boost::multiprecision::mpq_rational Type; };
-# else
+#if defined(CGAL_USE_BOOST_MP)
+{ typedef BOOST_gmp_arithmetic_kernel::Rational Type; };
+#else
 { typedef Gmpq Type; };
-# endif
+#endif
 #elif defined(CGAL_USE_LEDA)
 { typedef leda_rational Type; };
-#elif 0 && defined(CGAL_USE_BOOST_MP)
+#elif defined(CGAL_USE_BOOST_MP)
```

Comment:
So we enable it even for old boost? It wasn't really supported, so I guess that's ok.

Reply:
Some future version of boost will have a fast cpp_rational, and it makes perfect sense to enable it there. For users who only have an older version of boost, cpp_rational is rather slow, and it isn't clear if that is a better choice than the next option: `Quotient<MP_Float>` (bugs can also be a factor). But since this wasn't a supported case before, I am ok with your simple code that just always prefers boost.